src/components/country-list/country-list.component.js

```diff
@@ -0,0 +1,137 @@
+import { CovidDashboardService, COLOR_PALETTE } from "../../core/index";
+import { Chart } from "./chart/index";
+import { countryListTemplate } from "./country-list.template";
+import { Keyboard } from "../keyboard/index";
+import { createGlobalClick, clickChooseCountry } from "../util/index";
+
+export class CountryList {
+    constructor() {
+        this.chart = new Chart();
+        this.service = new CovidDashboardService();
+        this.titleCurrentCategory = "Cases";
+        this.paramNewOrTotal = "total";
+        this.param = `${this.paramNewOrTotal}${this.titleCurrentCategory}`;
+        this.counterSwitch = 0;
+        this.dataFullCountry = undefined;
+        this.data = undefined;
+        this.checkDataPopulations = false;
+        this.keyboardClass = false;
+    }
+
+    init() {
+        const list = document.querySelector(".country-list");
+        list.insertAdjacentHTML("beforeend", countryListTemplate);
+
+        this.initData();
+        Keyboard.init();
+    }
+
+    initData() {
+        this.service.getFullInformationCountry().then((data) => {
+            this.dataFullCountry = data;
+            this.data = data;
+            this.renderAndListen();
+        });
+    }
+
+    async renderAndListen() {
+        await this.render();
+        this.addEventListeners();
+    }
+
+    addEventListeners() {
+        // custom event
+        const countryList = document.querySelector(".country-list-wrapper");
+        const listWrapper = document.querySelector(".country-list-wrapper");
+        createGlobalClick(countryList);
+        clickChooseCountry(listWrapper);
+    }
+
+    render() {
+        this.chart.init(this.data);
+
+        setInterval(() => this.chart.search(), 1000);
+        const keyboardButton = document.querySelector(".country-list-keyboard");
+        const keyboard = document.querySelector(".keyboard");
+        keyboardButton.addEventListener("click", () => {
+            !this.keyboardClass
```

Comment:
Is it possible to use `toggle` method somehow `keyboard.classList.toggle("keyboard--hidden",this.keyboardClass )? It looks shorter but does it work?